src/module.cc

```diff
@@ -0,0 +1,388 @@
+    v8::V8::SetFlagsFromString("--abort_on_uncaught_exception", sizeof("--abort_on_uncaught_exception")-1);
+    isolate->SetAbortOnUncaughtExceptionCallback(OnUncaughtException);
+    exception_hook_initialised = true;
+  }
+
+#ifndef _WIN32
+  // If NodeReport newly requested on external user signal set up watchdog thread and handler
+  if ((nodereport_events & NR_SIGNAL) && (signal_thread_initialised == false)) {
+    SetupSignalHandler();
+  }
+  // If NodeReport no longer required on external user signal, reset the OS signal handler
+  if (!(nodereport_events & NR_SIGNAL) && (previous_events & NR_SIGNAL)) {
+    RestoreSignalHandler(nodereport_signal);
+  }
+#endif
+}
+NAN_METHOD(SetCoreDump) {
+  Nan::Utf8String parameter(info[0]);
+  nodereport_core = ProcessNodeReportCoreSwitch(*parameter);
+}
+NAN_METHOD(SetSignal) {
+#ifndef _WIN32
+  Nan::Utf8String parameter(info[0]);
+  unsigned int previous_signal = nodereport_signal; // save previous setting
+  nodereport_signal = ProcessNodeReportSignal(*parameter);
+
+  // If signal event active and selected signal has changed, switch the OS signal handler
+  if ((nodereport_events & NR_SIGNAL) && (nodereport_signal != previous_signal)) {
+    RestoreSignalHandler(previous_signal);
+    RegisterSignalHandler(nodereport_signal, SignalDump);
+  }
+#endif
+}
+NAN_METHOD(SetFileName) {
+  Nan::Utf8String parameter(info[0]);
+  ProcessNodeReportFileName(*parameter);
+}
+NAN_METHOD(SetDirectory) {
+  Nan::Utf8String parameter(info[0]);
+  ProcessNodeReportDirectory(*parameter);
+}
+NAN_METHOD(SetVerbose) {
+  Nan::Utf8String parameter(info[0]);
+  nodereport_verbose = ProcessNodeReportVerboseSwitch(*parameter);
+}
+
+/*******************************************************************************
+ * Callbacks for triggering NodeReport on failure events (as configured)
+ *  - fatal error
+ *  - uncaught exception
+ *  - signal
+ ******************************************************************************/
+static void OnFatalError(const char* location, const char* message) {
+  if (location) {
+    fprintf(stderr, "FATAL ERROR: %s %s\n", location, message);
+  } else {
+    fprintf(stderr, "FATAL ERROR: %s\n", message);
+  }
+  // Trigger NodeReport if requested
+  if (nodereport_events & NR_FATALERROR) {
+    TriggerNodeReport(Isolate::GetCurrent(), kFatalError, message, location, NULL);
+  }
+  fflush(stderr);
+  if (nodereport_core) {
+    raise(SIGABRT); // core dump requested (default)
+  } else {
+    exit(1); // user specified that no core dump is wanted, just exit
+  }
+}
+
+bool OnUncaughtException(v8::Isolate* isolate) {
+   // Trigger NodeReport if required
+  if (nodereport_events & NR_EXCEPTION) {
+    TriggerNodeReport(isolate, kException, "exception", "OnUncaughtException (nodereport/src/module.cc)", NULL);
+  } 
+  if (nodereport_core) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+#ifndef _WIN32
+static void SignalDumpInterruptCallback(Isolate *isolate, void *data) {
```

Comment:
Somewhat schizophrenic star style.  It leans to the right here but to the left in most other places.